server/sonar-qa-util/src/main/java/org/sonarqube/qa/util/pageobjects/RulesPage.java

```diff
@@ -21,46 +21,152 @@
+    for (String ruleName : ruleNames) {
+      getRulesCollection().findBy(text(ruleName)).shouldNotBe(visible);
+    }
+    return this;
+  }
+
+  public RulesPage shouldDisplayRuleWithLanguage(String ruleName, String languageName) {
+    getRulesCollection()
+      .findBy(Condition.and("", text(ruleName), text(languageName)))
+      .shouldBe(visible);
+    return this;
+  }
+
+  public RulesPage shouldNotDisplayRuleWithLanguage(String ruleName, String languageName) {
+    getRulesCollection()
+      .findBy(Condition.and("", text(ruleName), text(languageName)))
+      .shouldNotBe(visible);
     return this;
   }
 
   public RulesPage openFacet(String facet) {
-    Selenide.$(".search-navigator-facet-box[data-property=\"" + facet + "\"] .js-facet-toggle").click();
+    $(getFacetSelector(facet) + " .js-facet-toggle").click();
     return this;
   }
 
   public RulesPage selectFacetItemByText(String facet, String itemText) {
-    Selenide.$$(".search-navigator-facet-box[data-property=\"" + facet + "\"] .js-facet")
+    $$(getFacetSelector(facet) + " .js-facet")
       .findBy(Condition.text(itemText)).click();
     return this;
   }
 
+  public RulesPage selectFacetItem(String facet, String value) {
+    $(".search-navigator-facet-box[data-property=\"" + facet + "\"] .js-facet[data-value=\"" + value + "\"]").click();
+    return this;
+  }
+
+  public RulesPage selectInactive() {
+    $(getFacetSelector("qprofile") + " .search-navigator-facet.active .js-inactive").click();
+    return this;
+  }
+
+  public RulesPage shouldHaveDisabledFacet(String facet) {
+    $(".search-navigator-facet-box-forbidden[data-property=\"" + facet + "\"]").shouldBe(visible);
+    return this;
+  }
+
+  public RulesPage shouldNotHaveDisabledFacet(String facet) {
+    $(".search-navigator-facet-box-forbidden[data-property=\"" + facet + "\"]").shouldNotBe(visible);
+    return this;
+  }
+
   public RuleDetails openFirstRule() {
-    Selenide.$$(".js-rule").first().click();
-    Selenide.$(".coding-rules-details").shouldBe(Condition.visible);
+    $$(".js-rule").first().click();
+    $(".coding-rules-details").shouldBe(visible);
     return new RuleDetails();
   }
 
+  public RuleItem takeRuleByName(String ruleName) {
+    return new RuleItem(getRulesCollection().findBy(text(ruleName)));
+  }
+
+  public RuleItem takeRuleByName(String ruleName, int index) {
+    return new RuleItem(getRulesCollection().filterBy(text(ruleName)).get(index));
+  }
+
+  public RulesPage search(String query) {
+    $(".search-navigator-facet-query .search-box-input").val(query).pressEnter();
+    return this;
+  }
+
+  public RulesPage clearAllFilters() {
+    $(".js-new-search").click();
+    return this;
+  }
+
+  public RulesPage closeDetails() {
+    $(".js-back").click();
+    $(".coding-rules-details").shouldNotBe(visible);
+    return this;
+  }
+
+  public RulesPage activateOnlyRule() {
+    $$(".coding-rules-detail-quality-profile-activate").shouldHaveSize(1).first().click();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal ".coding-rules-detail-quality-profile-activate" 3 times. ([squid:S1192](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fit-rules&issues=AWEyHl7vzwlWAzALxAmZ&open=AWEyHl7vzwlWAzALxAmZ)

[ISSUE_KEY]: AWEyHl7vzwlWAzALxAmZ